app/src/main/java/com/example/habithelper/activities/MainActivity.java

```diff
@@ -1,41 +1,42 @@
-package com.example.habithelper;
+package com.example.habithelper.activities;
 
 import androidx.annotation.NonNull;
 import androidx.appcompat.app.AppCompatActivity;
 import androidx.fragment.app.Fragment;
 import androidx.fragment.app.FragmentManager;
 
-import android.content.Intent;
 import android.os.Bundle;
 import android.view.MenuItem;
-import android.view.View;
 import android.widget.Button;
 
+import com.example.habithelper.R;
 import com.example.habithelper.fragments.HomeFragment;
 import com.example.habithelper.fragments.ListFragment;
 import com.example.habithelper.fragments.MoodFragment;
 import com.example.habithelper.fragments.ProfileFragment;
 import com.example.habithelper.fragments.TrackFragment;
 import com.google.android.material.bottomnavigation.BottomNavigationView;
-import com.parse.ParseUser;
 
 public class MainActivity extends AppCompatActivity {
 
-    Button btnLogout;
     public static MainActivity self;
```

Comment:
having activity instance as public and static leads to memory leaks. It is a bad practice on android.

Reply:
I reference this instance in a fragment outside of the package, so that I can initialize an adapter in a fragment class. This is how they told us to perform that task during CodePath, but I agree that this seems like bad practice - do you have any ideas for ways I could still initialize an adapter from a fragment while not necessarily declaring the activity instance to have public access?

Reply:
In general all fragments have method get activity() to receive an activity they are associated with. But I am not sure what the challenge is with initializing an adapter in a fragment? What kind of adapter are we discussing? A recycler view adapter?